pkg/kubecost/allocation.go

```diff
@@ -1539,7 +1542,56 @@ func (as *AllocationSet) AggregateBy(aggregateBy []string, options *AllocationAg
 					if alloc.SharedCostBreakdown == nil {
 						alloc.SharedCostBreakdown = map[string]SharedCostBreakdown{}
 					}
-					sharedCostName := sharedAlloc.generateKey(aggregateBy, options.LabelConfig)
+
+					var sharedAggregateBy []string
+					for labelKey, _ := range options.SharedLabels {
+						sharedAgg := fmt.Sprintf("label:%s", labelKey)
+						if !slices.Contains(sharedAggregateBy, sharedAgg) {
+							sharedAggregateBy = append(sharedAggregateBy, sharedAgg)
+						}
+					}
+					if len(options.SharedNamespaces) > 0 {
+						sharedAggregateBy = append(sharedAggregateBy, "namespace")
+					}
+					sharedCostName := sharedAlloc.generateKey(sharedAggregateBy, options.LabelConfig)
+
+					// get each value in the generated key, then reset the name
+					sharedCostNameValues := strings.Split(sharedCostName, "/")
+					sharedCostName = ""
+
+					// try to match to the first label
+					for _, sharedLabelValues := range options.SharedLabels {
+						for _, sharedCostLabel := range sharedCostNameValues {
+							if slices.Contains(sharedLabelValues, sharedCostLabel) {
+								sharedCostName = sharedCostLabel
+								break
+							}
+						}
+						if len(sharedCostName) > 0 {
+							break
+						}
+					}
+
+					// if we didn't match to a label, try to match to a namespace
+					for _, sharedNamespace := range options.SharedNamespaces {
+						for _, sharedCostLabel := range sharedCostNameValues {
+							if sharedCostLabel == sharedNamespace {
```

Comment:
`sharedCostLabel` is a slightly misleading variable name, as it's not a label, but I don't think there's a logical issue here.

Reply:
Also, same issue here---you don't want to compare the shared namespaces to _all values_ here, right? Because some of those might be label values? We just want apples-to-apples comparisons.